### PR TITLE
Mccalluc/profile attribute error

### LINKED ISSUE #135: v0.0.13: AttributeError while running through Refinery
```
Traceback (most recent call last):
  File "/app_runner.py", line 98, in main
    server = init(args=args, parser=parser)
  File "/app_runner.py", line 38, in init
    if args.profile
AttributeError: 'RunnerArgs' object has no attribute 'profile'
```

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #135. At the root, the problem was that I was defining the "defaults" multiple times; I think this is better.

### CODE REVIEW

**Comment by mccalluc** [comment]
D'oh: Improved tests are looking for a root directory that might not be there. More mocks...

**Comment by mccalluc** [comment]
This PR improves coverage quite a bit, but one small gap that remains is downloading the files specified in the input.json. This is mocked in the new test, but we weren't anywhere close in the old tests.

File: context/app_runner_refinery.py [hunk 1]
```diff
@@ -75,24 +69,33 @@ def _download_files(self, urls, data_dir):
             else:
                 name = url.split("/")[-1]
                 path = '{}{}'.format(data_dir, name)
-                files.append(path)  # TODO: More unique?
                 with open(path, 'wb') as f:
                     for chunk in response.iter_content(chunk_size=1024):
                         # filter out KEEP-ALIVE new chunks
                         if chunk:
                             f.write(chunk)
+                files.append(open(path))
```
Comment (scottx611x) [inline context/app_runner_refinery.py]: Could we benefit here from:
```
with open(path) as f:
    files.append(f)
```
?

Comment (mccalluc) [reply context/app_runner_refinery.py]: In this case, no: When you specify that arguments are files in argparse, it actually gives you handles to the files... So we want to do the same thing here.

There is a [`close`](https://github.com/refinery-platform/heatmap-scatter-dash/pull/136/files#diff-511f078902f2e9d52889451a6ed43440R24) after the file is read.

(This is a change: [before](https://github.com/refinery-platform/heatmap-scatter-dash/pull/136/files#diff-511f078902f2e9d52889451a6ed43440L58) there was an if-then to take account of the different things that might be passed in.)

But thanks for noticing!

**Pull request merged**